includes/promoted-jobs/class-wp-job-manager-promoted-jobs-api.php

```diff
@@ -147,4 +172,28 @@ public function update_job_status( $request ) {
 			200
 		);
 	}
+
+	/**
+	 * Get the job data, but only if the user has permission to manage the job and the token is valid.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_REST_Response|WP_Error The response, or WP_Error on failure.
+	 */
+	public function get_job_data( $request ) {
+		$token    = $request->get_param( 'token' );
+		$user_id  = $request->get_param( 'user_id' );
+		$job_id   = $request->get_param( 'job_id' );
+		$verified = WP_Job_Manager_Site_Trust_Token::instance()->validate( 'user', $user_id, $token );
+		$result   = [
+			'verified' => $verified,
+		];
+		if ( $verified ) {
+			// We only want to return the job data if the specified user has permission to manage the job.
+			if ( ! user_can( $user_id, 'manage_job_listings', $job_id ) || 'job_listing' !== get_post_type( $job_id ) ) {
+				return new WP_Error( __( 'Job not found or does not have enough permissions.', 'wp-job-manager' ) );
```

Comment:
WDYT to add the status `401` here? Or separate and make a `404` too if the job is not found.

Reply:
Good call! Fixed in 4f7eca7 and 4529c2b. Thanks!